pkg/git/git.go

```diff
@@ -121,6 +130,79 @@ func Clone(
 	return r, r.clone()
 }
 
+// CopyRepo copies a git repository from the specified path to a temporary
+// location. Repository credentials are required in order to authenticate to the
+// remote repository, if any.
+func CopyRepo(path string, repoCreds RepoCredentials) (Repo, error) {
+	// Validate path is absolute
+	if !filepath.IsAbs(path) {
+		return nil, fmt.Errorf("path %s is not absolute", path)
+	}
+
+	// Validate path exists and is a directory
+	if fi, err := os.Stat(path); err != nil {
```

Comment:
Do we want to allow symlinks here?

Reply:
I think it may be ok as is:

https://go.dev/play/p/dh6OgL3sD0J

Reply:
Yes, that's what I expected.

My question was more if the current allowance of blindly following symlinks is intentional, as depending on where it is used it can result in e.g. path traversal attack vectors.

Reply:
(One fun fact to take into account is that a temporary directory created by Go on macOS is actually a symlink.)

Reply:
Think it is fine for now. We can potentially address this later.

---

My general recommendation is to always use `LStat ` unless you absolutely know you want to work with symlinks, to avoid hidden surprises. Even then I prefer to "follow" the symlink and work with the real path as much as possible at a high level to avoid having deal with the symlink throughout a code base.